src/ast/mod.rs

```diff
@@ -793,6 +793,11 @@ pub enum Expr {
     OuterJoin(Box<Expr>),
     /// A reference to the prior level in a CONNECT BY clause.
     Prior(Box<Expr>),
+    /// Scalar variable creation e.g. `[@]foo INT`
```

Comment:
Would it be possible to add a link to some documentation that uses this expression? I think that would be helpful in this case - e.g I'm unfamiliar with the syntax and unable to tell how it's used as a result